docs/edition_states.md

```diff
@@ -0,0 +1,75 @@
+# Edition States
+
+## Proposed State Diagram
+
+```mermaid
+flowchart TD
```

Comment:
this is super-useful. I've referred to it twice already today.

Just wondering why you opted to use the "flowchart" format rather than the Mermaid state diagram? https://mermaid.js.org/syntax/stateDiagram.html